src/components/App.tsx

```diff
@@ -547,31 +543,11 @@ class App extends React.Component<ExcalidrawProps, AppState> {
     }
 
     const scene = await loadScene(null, null, this.props.initialData);
-    this.props?.initializeScene?.(scene);
 
     if (this.state.isLoading) {
       this.setState({ isLoading: false });
     }
 
-    if (!this.props.initializeScene) {
-      this.setupScene(scene);
-    }
-
-    const addToLibraryUrl = new URLSearchParams(window.location.search).get(
-      "addLibrary",
-    );
-
-    if (addToLibraryUrl) {
-      await this.importLibraryFromUrl(addToLibraryUrl);
-    }
-  };
-
-  // can be renamed to something better ?
-  private setupScene = (scene: {
-    elements: readonly ExcalidrawElement[];
-    appState: MarkOptional<AppState, "offsetTop" | "offsetLeft">;
-    commitToHistory: boolean;
-  }) => {
```

Comment:
Removed. From now on the host app should supply `initialData` before rendering the compo (we could also consider allowing `initialData` to be a Promise).